test/common.py

```diff
@@ -232,15 +232,26 @@ def check_train_translate(train_params: str,
     # Run translate with restrict-lexicon
     data = run_translate_restrict(data, translate_params)
 
-    # Test scoring by ensuring that the sockeye.scoring module produces the same scores when scoring the output
-    # of sockeye.translate. However, since this training is on very small datasets, the output of sockeye.translate
-    # is often pure garbage or empty and cannot be scored. So we only try to score if we have some valid output
-    # to work with.
-    # Only run scoring under these conditions. Why?
-    # - translate splits up too-long sentences and translates them in sequence, invalidating the score, so skip that
-    # - scoring requires valid translation output to compare against
-    if '--max-input-len' not in translate_params and _translate_output_is_valid(data['test_outputs']):
-        test_scoring(data, translate_params, compare_translate_vs_scoring_scores)
+    # Test scoring. Inference does not guaranatee that </s> was generated, so we need to first
+    # force-decode against the output of the unconstrained system to get the real scores, that
+    # scoring will generate. Since it adds complexity to attach source factors to the input JSON
+    # object needed for constrained decoding, we only test scoring at the moment when not source
+    # factors are needed.
+    if 'test_source_factors' not in data:
+
+        # Run translate decoding against the reference
+        data = test_constrained_decoding_against_ref(data, translate_params)
```

Comment:
is this relevant for this PR? It seems that this is currently failing the system tests run with every commit on Travis.

Reply:
Yes, with that commit, we can simplify this.